pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -111,18 +113,93 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 // Name returns Name for the precondition.
 func (pf *Upgradeable) Name() string { return "ClusterVersionUpgradeable" }
 
+//  RecentEtcdBackup checks if a recent etcd backup has been taken.
+type RecentEtcdBackup struct {
+	key      string
+	lister   configv1listers.ClusterVersionLister
+	coLister configv1listers.ClusterOperatorLister
+}
+
+// NewRecentEtcdBackup returns a new RecentEtcdBackup precondition check.
+func NewRecentEtcdBackup(lister configv1listers.ClusterVersionLister, coLister configv1listers.ClusterOperatorLister) *RecentEtcdBackup {
+	return &RecentEtcdBackup{
+		key:      "version",
+		lister:   lister,
+		coLister: coLister,
+	}
+}
+
+func recentEtcdBackupCondition(lister configv1listers.ClusterOperatorLister) (string, string) {
+	var reason, message string
```

Comment:
nit: the function signature would be a bit more readable if we dropped these names in there, which would also allow you to drop this line:

```go
func recentEtcdBackupCondition(lister configv1listers.ClusterOperatorLister) (reason string, message string) {
```

Reply:
> which would also allow you to drop this line:

Not sure what you want? I can simply drop the function altogether and in-line it. Is that what you mean?